components/nimbus/src/tests/stateful/helpers.rs

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+use crate::{
+    glean_metrics::nimbus_events::{enrollment_status, EnrollmentStatusExtra},
+    metrics::{EnrollmentStatusExtraDef, MetricsHandler},
+};
+use glean::RecordedEvent;
+use std::sync::{Mutex, MutexGuard};
+
+impl From<EnrollmentStatusExtraDef> for EnrollmentStatusExtra {
+    fn from(enrollment_status_extra: EnrollmentStatusExtraDef) -> Self {
+        EnrollmentStatusExtra {
+            branch: enrollment_status_extra.branch,
+            conflict_slug: enrollment_status_extra.conflict_slug,
+            error_string: enrollment_status_extra.error_string,
+            reason: enrollment_status_extra.reason,
+            slug: enrollment_status_extra.slug,
+            status: enrollment_status_extra.status,
+        }
+    }
+}
+
+/// Implemented on RecordedEvents to aid in verifying the value of keys on event extras
+pub trait GetExtraKey {
+    fn get_extra_key(&self, key: &str) -> String;
+}
+
+impl GetExtraKey for RecordedEvent {
+    fn get_extra_key(&self, key: &str) -> String {
+        self.extra.clone().unwrap().get(key).unwrap().to_owned()
+    }
+}
+
+pub static METRICS_SINGLE_THREAD_MUTEX: Mutex<()> = Mutex::new(());
+
+pub fn reset_metrics<'a>() -> MutexGuard<'a, ()> {
+    let lock = match METRICS_SINGLE_THREAD_MUTEX.lock() {
+        Ok(l) => l,
+        Err(poisoned) => poisoned.into_inner(),
+    };
```

Comment:
This feels cursed but it works and I don't have to make a bunch of changes to the CI/builds to make sure it runs tests single-threaded